### PR TITLE
Simple map server + example client

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: examples/vmap/trillian_map_client/main.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"flag"
+	"log"
```
Comment (Martin2112) [inline examples/vmap/trillian_map_client/main.go]: glog? I've been using that everywhere.

File: server/vmap/trillian_map_server.go [hunk 2]
```diff
@@ -54,14 +56,153 @@ func (t *TrillianMapServer) getStorageForMap(mapId int64) (storage.MapStorage, e
...
+		return nil, err
+	}
+	rootHash, err := smtWriter.CalculateRoot()
+
+	newRoot := trillian.SignedMapRoot{
+		TimestampNanos: time.Now().UnixNano(),
+		RootHash:       rootHash,
+		MapId:          s.MapID().MapID,
+		MapRevision:    tx.WriteRevision(),
+		Signature:      &trillian.DigitallySigned{},
```
Comment (Martin2112) [inline server/vmap/trillian_map_server.go]: Should signature be empty? TODO?

File: examples/vmap/trillian_map_client/main.go (hunk 1 above)
Comment (AlCutter) [inline examples/vmap/trillian_map_client/main.go]: Done.

File: server/vmap/trillian_map_server.go (hunk 2 above)
Comment (AlCutter) [inline server/vmap/trillian_map_server.go]: Yeah, it's early days :)
Added a TODO.

**Pull request merged**